src/js_user.hpp

```diff
@@ -254,14 +256,43 @@ void UserClass<T>::link_credentials(ContextType ctx, ObjectType this_object, Arg
         }
 
         ValueType callback_arguments[2];
-        callback_arguments[0] = create_object<T, UserClass<T>>(ctx, new User<T>(shared_user, user->m_app));
+        callback_arguments[0] = create_object<T, UserClass<T>>(protected_ctx, new User<T>(shared_user, user->m_app));
         callback_arguments[1] = Value::from_undefined(protected_ctx);
         Function::callback(protected_ctx, protected_callback, typename T::Object(), 2, callback_arguments);
     });
 
     user->m_app->link_user(*user, credentials, callback_handler);
 }
 
+template<typename T>
+void UserClass<T>::call_function(ContextType ctx, ObjectType this_object, Arguments& args, ReturnValue &) {
+    args.validate_count(3);
+    auto user = get_internal<T, UserClass<T>>(ctx, this_object);
+
+    auto name = Value::validated_to_string(ctx, args[0], "name");
+    auto call_args_js = Value::validated_to_array(ctx, args[1], "args");
+    auto callback = Value::validated_to_function(ctx, args[2], "callback");
+    auto call_args_bson = Value::to_bson(ctx, call_args_js);
+
+    user->m_app->call_function(
+        *user,
```

Comment:
Probably make sense to validate user is not nullptr since get_internal will return that if `this` is not a valid js user instance.

Reply:
actually checked it. for node it will throw `invalid object` and for JSC it will return `nullptr`

Reply:
ok ignore this one. resolving